scully/plugins/algolia.ts

```diff
@@ -66,33 +78,35 @@ function buildPayload(options, mainTitle: string, content: string): Payload {
     };
 }
 
-const updateAlgoliaIndex = async (dom, options): Promise<JSDOM> => {
-    if (process.env.BUILD_ENV === 'develop' || options.route.startsWith('/reseller-support')) {
-        return dom;
-    }
+const updateAlgoliaIndex = async (routes: Array<HandledRoute>): Promise<void> => {
     try {
-        if (options.data?.title !== 'metadata' && options.data?.published && !options.data?.redirect) {
-            const { window } = dom;
-            const { document } = window;
-            const mainTitle = document.querySelector('h1')?.innerHTML || '';
-            const content = getPageContent(document);
-            const client = initAlgoliaClient();
-            const payload = buildPayload(options, mainTitle, content);
-
-            const indexToUse = client.initIndex(INDEX_NAME);
+        const client = initAlgoliaClient();
+
+        const indexToUse = client.initIndex(INDEX_NAME);
+
+        const objects = [];
+
+        for (const route of routes) {
+            if (
+                route.data?.title !== 'metadata' &&
+                route.data?.published &&
+                !route.data?.redirect &&
+                process.env.BUILD_ENV !== 'develop' &&
+                !route.route.startsWith('/reseller-support')
+            ) {
+                objects.push(await buildPayload(route));
```

Comment:
instead of await every buildPayload one by one, you can try to await all in scope
await Promise.all(objects)